msteams-platform/toolkit/provision.md

```diff
@@ -1,616 +1,436 @@
+* [Use an existing Azure AD app for your bot](#use-an-existing-azure-ad-app-for-your-bot)
 
-When the value is secret, then you don't need to hardcode them in parameter file. The parameter files support referencing the values from the environment variables. You can use this syntax `{{$env.YOUR_ENV_VARIABLE_NAME}}` in the parameter values for Teams Toolkit to resolve from the current environment variable.
+#### Use an existing Azure AD app for your Teams app
 
-The following example reads the value of the `mySelfHostedDbConnectionString` parameter from the environment variable `DB_CONNECTION_STRING`:
+You can follow below steps to add environment variables to the .env files to use an Azure AD app created for your Teams app. If you don't have an Azure AD app yet or you already have one but don't know where to find the correct value, see [how to use existing Azure AD app in TeamsFx project](https://github.com/devdiv-azure-service-dmitryr/teamsfx-docs/blob/main/V5-doc-update/use-existing-aad-app.md).
 
-```json
-...
-    "mySelfHostedDbConnectionString": "{{$env.DB_CONNECTION_STRING}}"
-...
-```
+1. Open `teamsapp.yml` and find the `aadApp/create` action.
 
-### Customize ARM template files
+1. Find the environment variable names that store information for Azure AD app in the `writeToEnvironmentFile` property. Below is the default `writeToenvironmentFile` definition if you create projects using Teams Toolkit:
 
-If the predefined templates don't meet your app requirements, you can customize the ARM templates under `templates\azure` folder. For example, you can customize the ARM template to create some extra Azure resources for your app. You need to have basic knowledge of bicep language, which is used to author ARM template.
+ ```typescript
+ writeToEnvironmentFile:
+   clientId: AAD_APP_CLIENT_ID
+   clientSecret: SECRET_AAD_APP_CLIENT_SECRET
+   objectId: AAD_APP_OBJECT_ID
+   tenantId: AAD_APP_TENANT_ID
+   authority: AAD_APP_OAUTH_AUTHORITY
+   authorityHost: AAD_APP_OAUTH_AUTHORITY_HOST
+ ```
 
-To ensure the TeamsFx tool functions properly, customize ARM template that satisfies the following requirements:
+1. Add values for each environment variable from step 2.
 
-* Ensure that the folder structure and file name remain unchanged. The tool may append new content to the existing files when you add more resources or capabilities to your project.
-* Ensure that the name of auto-generated parameters and its property names remain unhanged. The auto-generated parameters may be used when you add more resources or capabilities to your project.
-* Ensure that the output of auto-generated ARM template is unchanged. You can add more outputs to ARM template. The output is `.fx\states\state.{env}.json` and can be used in other features, such as deploy and validate manifest files.
+    1. Add below environment variables and their values to `env\.env.{env}` file.
 
-### Customize Teams app
+    ```typescript
```

Comment:
This is the codesnippet for .env file. AFAIK, markdown does not have syntax highlighting for .env file, so may delete the language part.

Reply:
Updated.